File: baekjoon/1197.py
from sys import stdin

v, e = map(int, stdin.readline().split())
edge = []
parent = [i for i in range(v+1)]

def find(a):
    if parent[a] == a:
        return a
    else:
        parent[a] = find(parent[a])
        return parent[a]

def union(a, b):
    a = find(a)
    b = find(b)
    if a != b:
        parent[b] = a

for _ in range(e):
    a, b, c = map(int, stdin.readline().split())
    edge.append((a, b, c))

edge.sort(key = lambda x : x[2]) # 세번째 값을 기준으로 정렬

usedEdge = 0
res = 0

for x in edge:
    if find(x[0]) != find(x[1]):
        union(x[0], x[1])
        usedEdge += 1
        res += x[2]
    if usedEdge >= v-1:
        break

print(res)

File: baekjoon/1197.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> parent;

int find(int a) {
    if (parent[a] == a) return a;
    else return parent[a] = find(parent[a]);
}

bool unite(int a, int b) {
    a = find(a);
    b = find(b);
    if (a == b) return false;
    parent[b] = a;
    return true;
}

void solve() {
    int v, e;
    cin >> v >> e;

    parent = vector<int>(v + 1);
    for (int i = 0; i < v + 1; i++) parent[i] = i;
    
    vector<vector<int>> edge;
    for (int i = 0; i < e; i++) {
        int a, b, c;
        cin >> a >> b >> c;
        edge.push_back({c, a, b});
    }

    sort(edge.begin(), edge.end());

    int ans = 0;
    int cnt = 0;
    for (auto x: edge) {
        if (unite(x[1], x[2])) {
            ans += x[0];
            cnt++;
        }
        if (cnt >= v - 1) break;
    }

    cout << ans << '\n';
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int T = 1;
    for (int i = 0; i < T; i++) {
        solve();
    }
}